compiler/rustc_middle/src/query/mod.rs

```diff
@@ -923,6 +931,10 @@ rustc_queries! {
     }
 
     Codegen {
+        // FIXME: remove after figuring out how to make miri able to detect non-Rust function calls
```

Comment:
I think you can check for `tcx.is_foreign(instance.def_id())`.

Reply:
Miri already detects those an shows an error, so I am confused by this comment actually.

Reply:
Miri uses `is_mir_available` for this as far as I know, which now checks for runtime mir instead of ctfe mir.

Reply:
That comment is outdated, as we also need this query for encoding MIR into metadata